lib/pluginmanager/bundler/logstash_injector.rb

```diff
@@ -93,6 +93,11 @@ def inject(gemfile_path, lockfile_path, dependencies)
 
           builder.eval_gemfile("bundler file", gemfile.generate())
           definition = builder.to_definition(lockfile_path, {})
+          # Remove the specific platform, and leave the generic platform
+          LogStash::Bundler.specific_platforms.each do |specific_platform|
```

Comment:
🤔 is there a reason we don't just invoke `LogStash::Bundler::genericise_platform` here? Is it possible that only removing the specific platforms and not adding a generic platform could leave us with no platforms?

Reply:
This particular code path is only used during the offline pack installation, and doesn't call out to the command-line bundler and the `invoke!` method that the `genericize_platform` uses. However, I have made a change here to call the `specific_platforms` filter on the platforms returned from `definition.platforms`, rather than making a call to `Gem::platforms`